### DISCUSSION TITLE
Adding express, fastify in Nuxt

### DISCUSSION THREAD

**Discussion by wobsoriano** [nuxt/nuxt#16365] [General]
Seeing a lot of questions regarding adding Express to Nuxt as a subsystem. Some were able to implement it, some were not.

Here's a demo of express, fastify and h3 routes in Nuxt implemented.

https://nuxt-subsystem-demo.vercel.app

Repo: https://github.com/wobsoriano/nuxt-subsystem-demo

Note that h3 supports connect/express middlewares and you don't even need to use express or fastify.

**Comment by jossephus** [comment]
Thank you for this!
can you please explain when u said "h3 supports connect/express middlewares"? and where can I read h3 documentation. I found `https://www.jsdocs.io/package/h3#getSession` but I think it is mostly for referencing not for learning.

**Reply by wobsoriano** [reply]
> h3 supports connect/express middlewares

h3 exports a `fromNodeMiddleware` function, which accepts a legacy express middleware. Say you want to use the [cors](https://expressjs.com/en/resources/middleware/cors.html) middleware:

```js
import { eventHandler, fromNodeMiddlware } from 'h3'
import cors from 'cors'

export default eventHandler(fromNodeMiddlware(cors()))
```


Check this [example](https://github.com/unjs/h3#more-app-usage-examples) for more info.
